server/pubsub/redis_query_results.go

```diff
@@ -188,20 +169,34 @@ func (r *redisQueryResults) ReadChannel(ctx context.Context, query fleet.Distrib
 	outChannel := make(chan interface{})
 	msgChannel := make(chan interface{})
 
+	conn := r.pool.Get()
```

Comment:
The docs mention the following: "For full concurrent access to Redis, use the thread-safe Pool to get, use and release a connection from within a goroutine." https://pkg.go.dev/github.com/garyburd/redigo/redis?utm_source=godoc#hdr-Concurrency

But we are now getting the connection outside of the goroutine.

Reply:
The current docs (https://pkg.go.dev/github.com/gomodule/redigo@v1.8.5/redis#hdr-Concurrency) are a little more specific:

> Connections support one concurrent caller to the Receive method and one concurrent caller to the Send and Flush methods. No other concurrency is supported including concurrent calls to the Do and Close methods. 

(`Unsubscribe` is a `Send` and `Flush` under the hood)

Reply:
I see. The `Get` part might still need to be in the same goroutine maybe?

Reply:
AFAICT the expected pattern is to open the conn and subscribe, then pass the connection to the separate goroutine. This is important as the call to `Unsubscribe` is what will get the `Receive` to return without a message and/or timeout. See the example https://pkg.go.dev/github.com/gomodule/redigo@v1.8.5/redis#example-PubSubConn